app/src/Containers/EventsContainer.tsx

```diff
@@ -118,16 +128,29 @@ const EventsContainer = (props: {
           )}
 
         <HorizontalLine />
-        <EventReturnedList data={data} />
+        <EventReturnedList data={data} eventsRange={selectedOption} />
+        
       </SafeArea>
     )
   }
 
-  if (allUpcomingEvents || eventsSearch) {
-    const eventsToShow = eventsSearch
-      ? eventsSearch.results
-      : (allUpcomingEvents as Events)
+  const [eventsToShow, setEventsToShow] = useState<Events>() 
+
+  useEffect (() => {
+    eventToShow()
+  }, [allUpcomingEvents, [props.route.params]])
```

Comment:
Hi Marcin, I've taken another look at this line as I didn't feel it was right.  The key thing here is to think about how `useEffect` works -- it runs its code in its first parameter (i.e. `eventToShow()`) any time one of the things in the array in its second parameter (line 141) changes.

If we look at the `eventToShow` function (lines 143-151), there are several variables that could affect the outcome of the logic of the `if` / `else if` statements here -- `eventsSearch`, `allUpcomingEvents`, `allPastEvents` and `selectedOption`.

Therefore, logically, we need to include all of these variables in the array that's the second parameter of `useEffect` on line 141 (in order to make `eventToShow()` run again) -- so this array should be `[eventsSearch, allUpcomingEvents, allPastEvents, selectedOption]`  I've also tried this out locally to check it works.

Hope that makes sense?

(I'm not sure quite what the effect was of putting `props.route.params` inside its own array -- `[props.route.params]` -- but it's definitely not the right answer here, even if it seemed to work in practice.  Always good to go back to first principles when something's unclear, to make sure we know why something should work.)

Reply:
Thanks for the insight Joe. I have replied to this message a day ago from my mobile when at work, however it did not show up here (I am guessing you have received an email with the reply which was not registered here). Thanks again for the detailed explanation.